### PR TITLE
migrate reporting to paging 3

### PR DESCRIPTION
**Pull request by connyduck**
this is where boilerplate code goes to die 😋

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusesPagingSource.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import kotlinx.coroutines.withContext
+
+class StatusesPagingSource(
+    private val accountId: String,
+    private val mastodonApi: MastodonApi
+) : PagingSource<String, Status>() {
+
+    override fun getRefreshKey(state: PagingState<String, Status>): String? {
+        state.anchorPosition.let { anchorPosition ->
+            return if (anchorPosition == null) {
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusesPagingSource.kt:34]: would it be easier to `?.let` ?

File: app/src/main/java/com/keylesspalace/tusky/components/report/fragments/ReportStatusesFragment.kt [hunk 2]
```diff
@@ -118,62 +118,46 @@ class ReportStatusesFragment : Fragment(R.layout.fragment_report_statuses), Inje
...
-            if (it?.status == com.keylesspalace.tusky.util.Status.RUNNING)
-                binding.progressBarTop.show()
-            else
-                binding.progressBarTop.hide()
-
-            if (it?.status == com.keylesspalace.tusky.util.Status.FAILED)
-                showError(it.msg)
-        }
+        adapter.addLoadStateListener { loadState ->
+            (loadState.refresh as? LoadState.Error?
```
Comment (charlag) [inline app/src/main/java/com/keylesspalace/tusky/components/report/fragments/ReportStatusesFragment.kt:135]: I would probably do `if` instead, it's a bit obfuscated

**Review by charlag — APPROVED** [review]
I'm not too familiar with either old or new API but looks good

File: app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusesPagingSource.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/report/adapter/StatusesPagingSource.kt:34]: yes definitely, 4 lines saved! thx

File: app/src/main/java/com/keylesspalace/tusky/components/report/fragments/ReportStatusesFragment.kt (hunk 2 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/report/fragments/ReportStatusesFragment.kt:135]: good point

**Pull request merged**